src/index.js

```diff
@@ -184,14 +198,18 @@ export class SlideDeck extends React.Component {
                 {...this.state}
                 slides={slides}
                 width={width}
-                height={height}>
+                height={height}
+                update={this.update}>
                 <GoogleFonts />
                 <Carousel index={index}>
                   {slides.map((Component, i) => (
                     <Slide
                       key={i}
                       id={'slide-' + i}
-                      index={i}>
+                      index={i}
+                      active={i === index}
+                      update={this.update}
```

Comment:
I think this is a little confusing as to what's happening here? This would provide the main app's state update methods in two different contexts now, right?

Reply:
What other context would we have available? Might have missed something here.

Reply:
Yeah, sorry this is moving around quickly, the main app uses this context module https://github.com/jxnblk/mdx-deck/blob/master/src/context.js here https://github.com/jxnblk/mdx-deck/blob/master/src/index.js#L175

Reply:
Can `import { withDeck } from './context'` to hook into it

Reply:
The slide context is because the notes need to know which slide index to appear in – there might be a way to consolidate this all into the slide context tbh